integrations/oci/oci-base/src/main/java/io/helidon/integrations/oci/Oci.java

```diff
@@ -0,0 +1,335 @@
+                    } else {
+                        temp = new ConfigFileAuthenticationDetailsProvider(ociConfigPath, ociAuthProfile);
+                    }
+                } catch (final IOException ioException) {
+                    temp = null;
+                } finally {
+                    return temp;
+                }
+            } else { //Should we use MANUAL here?
+                SimpleAuthenticationDetailsProvider simpleProvider =
+                        SimpleAuthenticationDetailsProvider.builder()
+                                .tenantId(ociAuthTenancy)
+                                .userId(ociAuthUser)
+                                .fingerprint(ociAuthFingerprint)
+                                .region(Region.valueOf(ociAuthRegion))
+                                .passphraseCharacters(ociAuthPassphraseCharacters.toCharArray())
+                                .privateKeySupplier(this::getPrivateKey)
+                                .build();
+                return simpleProvider;
+            }
+        }
+
+        private ClientConfiguration configureClient() {
+            ClientConfiguration clientConfig
+                    = ClientConfiguration.builder()
+                    .connectionTimeoutMillis(clientConnectionTimeoutMillis)
+                    .readTimeoutMillis(clientReadTimeoutMillis)
+                    .build();
+            return clientConfig;
+        }
+
+        /**
+         * Access point to the provider.
+         *
+         * @return AuthenticationDetailsProvider as the main entry point.
+         */
+        public AuthenticationDetailsProvider getProvider() {
+            return provider;
+        }
+
+        /**
+         * Access point to the Client configuration.
+         *
+         * @return ClientConfiguration built and configured.
+         */
+        public ClientConfiguration getClientConfig() {
+            return clientConfig;
+        }
+
+        private InputStream getPrivateKey() {
+            if (ociAuthPrivateKey == null || ociAuthPrivateKey.trim().isEmpty()) {
+                final String pemFormattedPrivateKeyFilePath =
+                        Optional.ofNullable(ociAuthKeyFile)
+                                .orElse(Paths.get(System.getProperty("user.home"), ".oci/oci_api_key.pem").toString());
+                assert pemFormattedPrivateKeyFilePath != null;
+                try {
+                    return new BufferedInputStream(Files.newInputStream(Paths.get(pemFormattedPrivateKeyFilePath)));
+                } catch (final IOException ioException) {
+                    throw new RuntimeException(ioException.getMessage(), ioException);
+                }
+            } else {
+                return new BufferedInputStream(new ByteArrayInputStream(ociAuthPrivateKey.getBytes(StandardCharsets.UTF_8)));
+            }
+        }
+
+        private Builder ociAuthProfile(String ociAuthProfile) {
```

Comment:
These methods should be public and documented

Reply:
And it should have the enum as a parameter

Reply:
fixed